packages/core-p2p/lib/monitor.js

```diff
@@ -103,47 +103,33 @@ class Monitor {
     }
 
     if (networkStart) {
-      logger.warn(
-        'Skipped peer discovery because the relay is in genesis-start mode.',
-      )
+      logger.warn('Skipped peer discovery because the relay is in genesis-start mode.')
       return
     }
 
     if (this.config.disableDiscovery) {
-      logger.warn(
-        'Skipped peer discovery because the relay is in non-discovery mode.',
-      )
+      logger.warn('Skipped peer discovery because the relay is in non-discovery mode.')
       return
     }
 
     try {
-      if (process.env.ARK_ENV !== 'test') {
-        await this.discoverPeers()
-        await this.cleanPeers()
-
-        if (!this.hasMinimumPeers()) {
-          this.__addPeers(config.peers.list)
-
-          logger.info("Couldn't find enough peers, trying again in 5 seconds.")
-
-          await delay(5000)
-
-          return this.updateNetworkStatus()
-        }
-      }
+      await this.discoverPeers()
+      await this.cleanPeers()
     } catch (error) {
       logger.error(`Network Status: ${error.message}`)
+    }
 
-      this.__addPeers(config.peers.list)
+    let nextRunDelaySeconds = 60
 
-      logger.info('Failed to discover peers, trying again in 5 seconds.')
+    if (!this.hasMinimumPeers()) {
+      this.__addPeers(config.peers.list)
 
-      if (process.env.NODE_ENV !== 'test') {
-        await delay(5000)
-      }
+      nextRunDelaySeconds = 5
 
-      return this.updateNetworkStatus()
+      logger.info(`Couldn't find enough peers, trying again in ${nextRunDelaySeconds} seconds`)
     }
+
+    setTimeout(() => await this.updateNetworkStatusIfNotEnoughPeers(), nextRunDelaySeconds * 1000)
```

Comment:
This also needs to be wrapped into the check from above. Unit tests are not supposed to to call it.

Reply:
Or wait, probably wrong what I just said. You are missing an `async` .

Reply:
> This also needs to be wrapped into the check from above. Unit tests are not supposed to call it.

No because we do `return;` if we detect a test environment at the start of the function.

> You are missing an async .

Where?